src/components/FilteredProducts/FilteredProducts.jsx

```diff
@@ -0,0 +1,33 @@
+import { useState } from "react";
+import Card from "../../components/Card";
+import Loading from "../Loading";
+
+const FilteredProducts = ({ selectedProducts }) => {
+  const [isLoading, setIsLoading] = useState(true);
+  setTimeout(() => {
+    setIsLoading(false);
+  }, 2000);
+
+  if (isLoading) {
+    return <Loading />
+  }
+  return (
+    <div>
+      {selectedProducts.length > 0 ? (
+        selectedProducts.map((product) => (
+          <Card
+            key={product.id}
+            name={product.data.name}
+            image={product.data.mainimage.url}
+            isProduct={true}
+            price={product.data.price}
+          />
+        ))
+      ) : (
+        <p>No products</p>
+      )}
+    </div>
+  );
```

Comment:
Algo que en lo personal me gusta en lugar de ocupar un ternario es un early return como en tu loading
```suggestion
  if (selectedProducts.length > 0) {
    return <p>No products</p>
  }
  return (
    <div>
        selectedProducts.map((product) => (
          <Card
            key={product.id}
            name={product.data.name}
            image={product.data.mainimage.url}
            isProduct={true}
            price={product.data.price}
          />
        ))
    </div>
  );
};
```